code/game/objects/items.dm

```diff
@@ -871,6 +871,82 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 		return
 	usr.UnarmedAttack(src)
 
+/obj/item/proc/use_tool(atom/target, mob/living/user, delay, amount = 0, volume = 0, datum/callback/extra_checks)
+	// No delay means there is no start message, and no reason to call tool_start_check before use_tool.
+	// Run the start check here so we wouldn't have to call it manually.
+
+	//if(user.is_busy()) to be ported in future if we ever need it
+	//	return
+
+	if(!delay && !tool_start_check(user, amount))
+		return
+
+	delay *= toolspeed
```

Comment:
```suggestion
	delay /= toolspeed
```
TG's toolspeed is smaller = better, ours is larger = better, so right now in this PR the impact driver takes 3x as long to do anything